### ISSUE TITLE
panic: assignment to entry in nil map

### ISSUE THREAD

**Issue by fkondej** [vegaprotocol/vega#6438]
# Problem encountered
A panic happened on a non-validator node on the Fairground network, and it caused the node to crash, which then restarted with a visor.

# Observed behaviour
A non-validator node restarted from block 0

# Expected behaviour
The node should not panic/crash.

# Evidence

## Logs
```
panic: assignment to entry in nil map
goroutine 22909253 [running]:
code.vegaprotocol.io/vega/datanode/utils.(*Observer[...]).Subscribe(0xc000306548, {0x2acf220, 0xc00369edc0}, 0xc00269f3e0)
        /home/runner/work/vega/vega/datanode/utils/observer.go:57 +0xdc
code.vegaprotocol.io/vega/datanode/utils.(*Observer[...]).Observe(0xc000306548, {0x2acf220, 0xc00369edc0}, 0x3, 0xc00445d1b8)
        /home/runner/work/vega/vega/datanode/utils/observer.go:114 +0x7c
code.vegaprotocol.io/vega/datanode/sqlstore.(*LiquidityProvision).ObserveLiquidityProvisions(0xc000306510, {0x2acf220, 0xc00369edc0}, 0xc00462ff40?, 0x0, 0x0)
        /home/runner/work/vega/vega/datanode/sqlstore/liquidity_provision.go:65 +0xb1
code.vegaprotocol.io/vega/datanode/api.(*tradingDataServiceV2).ObserveLiquidityProvisions(0xc001072000, 0xc00462ff40, {0x2ad8990, 0xc00269f3d0})
        /home/runner/work/vega/vega/datanode/api/trading_data_v2.go:1752 +0xcb
code.vegaprotocol.io/vega/protos/data-node/api/v2._TradingDataService_ObserveLiquidityProvisions_Handler({0x247b400?, 0xc001072000}, {0x2ad60c0, 0xc001ce20d0})
        /home/runner/work/vega/vega/protos/data-node/api/v2/trading_data_grpc.pb.go:2800 +0xd3
google.golang.org/grpc.(*Server).processStreamingRPC(0xc001d48000, {0x2add100, 0xc0010a2000}, 0xc0049efb00, 0xc0014e2bd0, 0x3d28620, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1558 +0xfc7
google.golang.org/grpc.(*Server).handleStream(0xc001d48000, {0x2add100, 0xc0010a2000}, 0xc0049efb00, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1640 +0x9ea
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:932 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:930 +0x28a
```

to view logs:
```bash
ssh n06.testnet.vega.xyz
sudo journalctl -u vegavisor --since "2022-10-06 16:15:00" --until "2022-10-06 16:16:00"
```

## Additional context
v0.57.0 - run with visor.

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

**Comment by gordsport** [comment]
Closing as fixed already but need 0.58 with the LP subscription fix

**Comment by gordsport** [comment]
Fixed in 

https://github.com/vegaprotocol/vega/pull/6409
